pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/MaxAggregationFunction.java

```diff
@@ -190,12 +190,28 @@ public void aggregateGroupBySV(int length, int[] groupKeyArray, GroupByResultHol
   @Override
   public void aggregateGroupByMV(int length, int[][] groupKeysArray, GroupByResultHolder groupByResultHolder,
       Map<ExpressionContext, BlockValSet> blockValSetMap) {
+    BlockValSet blockValSet = blockValSetMap.get(_expression);
     double[] valueArray = blockValSetMap.get(_expression).getDoubleValuesSV();
-    for (int i = 0; i < length; i++) {
-      double value = valueArray[i];
-      for (int groupKey : groupKeysArray[i]) {
-        if (value > groupByResultHolder.getDoubleResult(groupKey)) {
-          groupByResultHolder.setValueForKey(groupKey, value);
+
+    if (_nullHandlingEnabled) {
+      forEachNotNull(length, blockValSet, (from, to) -> {
```

Comment:
nit: given you already have the BlockValSet in `blockValSet` you can just slightly simplify 
```
    double[] valueArray = blockValSetMap.get(_expression).getDoubleValuesSV();
```